xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/migration/R150000000XWIKI20285DataMigration.java

```diff
@@ -73,22 +76,26 @@ protected String getTaskType()
     }
 
     @Override
-    protected List<String> selectDocuments() throws DataMigrationException
+    protected List<ReferenceWithLocale> selectDocuments() throws DataMigrationException
     {
         XWiki wiki = getXWikiContext().getWiki();
         try {
             // We select potentially impacted documents using like wildcards. This selection might lead to false 
             // positive that wild be filtered out by the more accurate regex in
             // InvitationInternalDocumentParameterEscapingFixer.
             return wiki.getStore().getQueryManager()
-                .createQuery("where doc.content "
+                .createQuery("SELECT doc.fullName, doc.language from Document doc where doc.content "
                     + "like '%{{info}}%services.localization.render%xe.invitation.internalDocument%{{/info}}%'", XWQL)
-                .<String>execute()
+                .<Object[]>execute()
                 .stream()
+                .map(array -> new ReferenceWithLocale(String.valueOf(array[0]), String.valueOf(array[1])))
                 // Exclude document that are provided by extensions, because they are fixed using the usual xar upgrade
                 // mechanism.
-                .filter(documentReference -> !this.installedXARs.isExtensionDocument(
-                    this.documentReferenceResolver.resolve(documentReference)))
+                .filter(documentReference -> {
+                    DocumentReference resolve = this.documentReferenceResolver.resolve(documentReference.reference);
+                    return !this.installedXARs.isExtensionDocument(
+                        new DocumentReference(resolve, toLocale(documentReference.locale, ROOT)));
```

Comment:
Same potentially unsupported locale problem as the one discussed above.

Reply:
Actually, I'm wondering if it wouldn't be cleaner to make `selectDocuments` return a List<DocumentReference>, that would avoid having to resolve the reference twice in the case of R150000000XWIKI20285DataMigration (and it makes more sense as an API, even if AbstractDocumentsMigration is internal).